slang-gfx.h

```diff
@@ -436,6 +436,14 @@ class IResource: public ISlangUnknown
         CountOf,
     };
 
+    enum NativeHandleType
```

Comment:
It did occur to me yesterday that this might be the same thing as `DeviceType` since it stands to reason that the source API for a native handle ought to be the same as the API of the source device that's providing it (and `NativeHandleType` as is looks suspiciously like a subset of `DeviceType`). In hindsight, this enum is poorly named and should probably be something like `HandleSourceAPI`.

I do wonder if maybe it'd be better to at least merge native and shared handles into a single struct though. Perhaps like
```
struct ResourceHandles
{
    uint64_t nativeHandle = 0;
    void* sharedHandle = nullptr;
    DeviceType srcApi = DeviceType::Unknown;
    Type type = Type::Unknown; // This is the enum class under IResource
```
This would let us track all the handles for a single resource in one place, and the getters would just fill in the appropriate fields. Though having written this, I think I should probably give `SharedHandle` the same treatment as `NativeResourceHandle`...

Reply:
I am thinking of something like
```
enum class InteropHandleType
{
     Unknown, D3D12, Vulkan, CUDA, Win32, FileDescriptor
};

struct InteropHandle
{
     InteropHandleType type;
     uint64_t handleValue = 0;
};
```

Reply:
So everything would basically store two instances of this struct, one for the native handle and one for the shared handle? I still think that it should also track whether the handle is for a buffer/texture/etc though, especially if we also take Tess's suggestion to have a single unified struct for handles instead of having resource handles use a separate struct from device handles.

Reply:
I didn't quite get on why we need to track if a handle is for buffer or texture? I'd assume it is a user's error error if they attempt to use the shared handles with mismatching types.